python/s3daemon/s3daemon.py

```diff
@@ -61,30 +63,29 @@
 log.setLevel(logging.INFO)
 
 
-async def handle_client(client, reader, writer):
+async def handle_client(client, conn):
     """Handle a client connection to the server socket.
 
     Parameters
     ----------
     client : `S3`
         The S3 client to use to talk to the server.
-    reader : `asyncio.StreamReader`
-        A stream connected to the socket to read the filename/destination pair.
-    writer : `asyncio.StreamWriter`
-        A stream connected to the socket to write back status information.
+    conn : `socket.Socket`
+        The socket connected to the client.
     """
-    filename, dest = (await reader.readline()).decode("UTF-8").rstrip().split(" ")
+    filename, dest = conn.recv(4096).decode("UTF-8").rstrip().split(" ")
     start = time.time()
     # ignore the alias
     _, bucket, key = dest.split("/", maxsplit=2)
     try:
         with open(filename, "rb") as f:
             await client.put_object(Body=f, Bucket=bucket, Key=key)
-            writer.write(b"Success")
-            log.info("%f %f sec - %s", start, time.time() - start, filename)
+            conn.send(b"Success")
+            log.info("%f %f sec - %s", start, time.time() - start, key)
     except Exception as e:
-        writer.write(bytes(repr(e), "UTF-8"))
-        log.exception("%f %f sec - %s", start, time.time() - start, filename)
+        conn.send(bytes(repr(e), "UTF-8"))
```

Comment:
```suggestion
        conn.send(repr(e).encode())
```
? or `conn.send(repr(e).encode("UTF-8"))`?